WalletWasabi.Fluent/Controls/MultiSelector.axaml.cs

```diff
@@ -0,0 +1,90 @@
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Reactive;
+using System.Reactive.Disposables;
+using System.Reactive.Linq;
+using System.Reactive.Subjects;
+using Avalonia;
+using Avalonia.Controls;
+using DynamicData;
+using DynamicData.Binding;
+using ReactiveUI;
+using WalletWasabi.Fluent.Extensions;
+
+namespace WalletWasabi.Fluent.Controls;
+
+public class MultiSelector : ItemsControl, IDisposable
+{
+    private readonly CompositeDisposable _disposables = new();
+
+	[System.Diagnostics.CodeAnalysis.SuppressMessage("Reliability", "CA2000:Dispose objects before losing scope", Justification = "Using DisposeWith")]
+	public MultiSelector()
+    {
+        var changes = this
+	        .WhenAnyValue(x => x.Items)
+	        .WhereNotNull()
+	        .OfType<ReadOnlyObservableCollection<ISelectable>>()
+	        .Select(x => x.ToObservableChangeSet())
+	        .Switch();
+
+		var isSelected = changes
+            .AutoRefresh(x => x.IsSelected)
+            .ToCollection()
+            .Select(GetSelectionState);
+
+        var isSelectedSubject = new BehaviorSubject<bool?>(false);
+        isSelected.Subscribe(isSelectedSubject);
+        
+        Toggle = ReactiveCommand.CreateFromObservable(() =>
+            {
+                var currentValue = isSelectedSubject.Value;
+                return ToggleChildrenSelection(changes, GetNextValue(currentValue));
+            })
+            .DisposeWith(_disposables);
+
+        IsChecked = isSelected
+	        .CombineLatest(Toggle.IsExecuting)
+	        .Where(x => !x.Second)
+	        .Select(x => x.First)
+	        .ReplayLastActive();
+    }
+
+    public ReactiveCommand<Unit, IReadOnlyCollection<ISelectable>> Toggle { get; }
+
+    public IObservable<bool?> IsChecked { get; }
+
+    public void Dispose()
+    {
+        _disposables.Dispose();
+    }
+
+    protected override void OnDetachedFromVisualTree(VisualTreeAttachmentEventArgs e)
+    {
+        _disposables.Dispose();
+        base.OnDetachedFromVisualTree(e);
+    }
```

Comment:
Yeah, but what proves a `MultiSelector` won't be used anywhere else in the future? Where a theme change is possible during usage.

Reply:
You're right, I'll tackle this last because it's the less important.

Reply:
> AFAIK it can be detached and reattached many times. It would throw an exception in that case. I think if you move the logic and the `_disposables` init to the `OnAttached` method, that can solve the problem. Is it possible?

I have made tests and I found out that theme switching creates a new instance, so this design is safe. 

I won't modify it.